main.go

```diff
@@ -0,0 +1,393 @@
+			glog.Fatalf("Error in rate limiter for user %s %v", u.PrimaryEmail, err)
+			return err
+		}
+		if ctx.Err() != nil {
+			glog.Fatalf("Error in rate limiter for user %s %v", u.PrimaryEmail, ctx.Err())
+			return ctx.Err()
+		}
+		return nil
+	}); err != nil {
+		glog.Errorf("Error iterating visible organizations for user %s %v", u.PrimaryEmail, err)
+		return []*cloudresourcemanager.Organization{}, err
+	}
+
+	for _, o := range organizations {
+		glog.V(50).Infof("             User %s has Organiation visibility to %s", u.PrimaryEmail, o.Name)
+		if _, ok := allOrganizations[o.Name]; !ok {
+			glog.V(2).Infof("             User [%s] has external organization visibility to [%s](%s)", u.PrimaryEmail, o.Name, o.DisplayName)
+		}
+	}
+	return organizations, nil
+}
+
+func getProjects(ctx context.Context, u admin.User) ([]*cloudresourcemanager.Project, error) {
+	glog.V(50).Infof("             Getting Projects for user %s", u.PrimaryEmail)
+	defer wg.Done()
+
+	var crmService *cloudresourcemanager.Service
+
+	if *impersonatedServiceAccount != "" {
+		ts, err := impersonate.CredentialsTokenSource(ctx, impersonate.CredentialsConfig{
+			TargetPrincipal: *impersonatedServiceAccount,
+			Scopes:          []string{cloudresourcemanager.CloudPlatformReadOnlyScope},
+			Subject:         u.PrimaryEmail,
+		})
+		if err != nil {
+			glog.Errorf("Error creating CRM credentials for user %s,  %v", u.PrimaryEmail, err)
+			return []*cloudresourcemanager.Project{}, err
+		}
+		crmService, err = cloudresourcemanager.NewService(ctx, option.WithTokenSource(ts))
+		if err != nil {
+			glog.Errorf("Error creating CRM Service for  user %s,  %v", u.PrimaryEmail, err)
+			return []*cloudresourcemanager.Project{}, err
+		}
+	} else {
+		cred, err := google.CredentialsFromJSONWithParams(ctx, svcAccountJSONBytes, google.CredentialsParams{
+			Scopes:  []string{cloudresourcemanager.CloudPlatformReadOnlyScope},
+			Subject: u.PrimaryEmail,
+		})
+		if err != nil {
+			glog.Errorf("Error creating CRM credentials for  user %s,  %v", u.PrimaryEmail, err)
+			return []*cloudresourcemanager.Project{}, err
+		}
+		crmService, err = cloudresourcemanager.NewService(ctx, option.WithCredentials(cred))
+		if err != nil {
+			glog.Errorf("Error creating CRM Service for  user %s,  %v", u.PrimaryEmail, err)
+			return []*cloudresourcemanager.Project{}, err
+		}
+	}
+
+	projects := make([]*cloudresourcemanager.Project, 0)
+	// do not apply a list filter here; we need to audit all projects
+	//  https://pkg.go.dev/google.golang.org/api@v0.58.0/cloudresourcemanager/v1#ProjectsListCall.Filter
+	noFilter := ""
+
+	req := crmService.Projects.List().Filter(noFilter).PageSize(maxPageSize)
+	if err := req.Pages(ctx, func(page *cloudresourcemanager.ListProjectsResponse) error {
+		projects = append(projects, page.Projects...)
+		if err := limiter.Wait(ctx); err != nil {
+			glog.Errorf("Error in rate limiter for user %s %v", u.PrimaryEmail, err)
+			return err
+		}
+		if ctx.Err() != nil {
```

Comment:
You shouldn't need to check the context error here, the error returned from limiter should work.